projects/igniteui-angular/src/lib/grids/filtering/excel-style/excel-style-search.component.ts

```diff
@@ -46,17 +46,18 @@ export class IgxExcelStyleSearchComponent {
 
     public onCheckboxChange(eventArgs: IChangeCheckboxEventArgs) {
         const selectAll = this.column.grid.resourceStrings.igx_grid_excel_select_all;
-        if (eventArgs.checkbox.value.value === selectAll) {
+        const selectedIndex = this.data.indexOf(eventArgs.checkbox.value);
+        if (eventArgs.checkbox.value.value === selectAll && selectedIndex === 0) {
             this.data.forEach(element => {
                 element.isSelected = eventArgs.checked;
                 this.data[0].indeterminate = false;
             });
         } else {
             eventArgs.checkbox.value.isSelected = eventArgs.checked;
-            if (!this.data.filter(el => el.value !== selectAll).find(el => el.isSelected === false)) {
+            if (!this.data.filter(el => selectedIndex !== 0).find(el => el.isSelected === false)) {
                 this.data[0].indeterminate = false;
                 this.data[0].isSelected = true;
-            } else if (!this.data.filter(el => el.value !== selectAll).find(el => el.isSelected === true)) {
+            } else if (!this.data.filter(el => selectedIndex !== 0).find(el => el.isSelected === true)) {
```

Comment:
You can use `this.data.slice(1, this.data.length)` instead of `this.data.filter(el => selectedIndex !== 0)` similar to what you're doing in the next file. It would be much faster for large arrays.